### PR TITLE
BTA-1005 VAT payment cards

### PR DESCRIPTION
**Pull request by gianick**

### CODE REVIEW

File: app/services/VatCardBuilderService.scala [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        case VatUnactivated  => ???
+        case data: VatData   => buildVatCardData(
+                                  paymentsContent = Some(vatPartialBuilder.buildPaymentsPartial(data).toString()),
+                                  returnsContent = Some(vatPartialBuilder.buildReturnsPartial(data, request.vatDecEnrolment).toString())
+                                )
+      }
+    } recover {
+      case e: Throwable      => ???
+    }
+  }
```
Comment (AndrewShanks) [inline app/services/VatCardBuilderService.scala]: There are a lot of unimplemented branches here.

File: app/services/VatCardBuilderService.scala [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                                  returnsContent = Some("")
+                                )
+        case VatEmpty        => ???
+        case VatUnactivated  => ???
+        case data: VatData   => buildVatCardData(
+                                  paymentsContent = Some(vatPartialBuilder.buildPaymentsPartial(data).toString()),
+                                  returnsContent = Some(vatPartialBuilder.buildReturnsPartial(data, request.vatDecEnrolment).toString())
+                                )
+      }
+    } recover {
```
Comment (AndrewShanks) [inline app/services/VatCardBuilderService.scala]: Probably better to recover at the controller level to throw a 500 to BTA which will display the error card

File: app/services/VatPartialBuilder.scala [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+          else {
+            Html("")
+          }
+        }
+        case _ => views.html.partials.vat.card.payments.payments_fragment_no_data()
+      }
+      case VatNoData => views.html.partials.vat.card.payments.payments_fragment_no_data()
+      case VatGenericError => Html("")
+      case VatEmpty => Html("")
+      case VatUnactivated => Html("")
```
Comment (AndrewShanks) [inline app/services/VatPartialBuilder.scala]: The partial builder is distinguishing between VatData and other statuses when the card builder already does that. One of these is redundant.

Comment (gianick) [reply app/services/VatPartialBuilder.scala]: Refactored VatCardBuilderService and VatpartialBuilder. Now the VatPartialBuilder caters only for the cases when we have VatData and the others are handled in the CardBuilderService.

File: app/services/VatCardBuilderService.scala (hunk 2 above)
Comment (gianick) [reply app/services/VatCardBuilderService.scala]: Removed the "recover". it's now handled by the PartialController.

File: app/services/VatCardBuilderService.scala (hunk 1 above)
Comment (gianick) [reply app/services/VatCardBuilderService.scala]: Left them in order to have them implemented as part of a future story.

**Pull request merged**